app/scripts/modules/core/presentation/HoverablePopover.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import {Overlay, Popover} from 'react-bootstrap';
+import {UUIDGenerator} from '../utils/uuid.service';
+
+interface IState {
+  popoverIsOpen: boolean;
+  target?: any;
+  popoverCancel?: number;
+}
+
+interface IProps {
+  value?: string;
+  template?: JSX.Element;
+  placement?: 'top' | 'bottom' | 'left' | 'right';
+  id?: string;
+}
+
+export class HoverablePopover extends React.Component<IProps, IState> {
+  public static defaultProps: Partial<IProps> = {
+    placement: 'top',
+    id: UUIDGenerator.generateUuid()
+  };
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      popoverIsOpen: false,
+    };
+  }
+
+  public componentWillUnmount() {
+    this.hidePopover();
```

Comment:
this will probably throw up because there's a `setState` in `hidePopover()`. Should just call `this.clearPopoverCancel();` Wait, no can't do that, there's a setState in there too.

Now that I look more closely, `popoverCancel` probably shouldn't be on state. Should just be an instance variable, like the rx Subscriptions elsewhere in the code.